File: 프로그래머스/lv3/42628. 이중우선순위큐/이중우선순위큐.js
function solution(operations) {
    const q = []
    operations.forEach(operation => {
        let [command, number] = operation.split(" ")
        number = parseInt(number)
        
        switch(command) {
            case "I":
                q.push(number)
                break
            case "D":
                if (q.length === 0) break
                if (number === 1) {
                    q.sort((a, b) => a - b)
                } else if (number === -1) {
                    q.sort((a, b) => b - a)
                }
                q.pop()
                break
        }
    })
    
    if (q.length) {
        q.sort((a, b) => a - b)
        return [q[q.length - 1], q[0]]
    } else return [0, 0]
}

File: 프로그래머스/lv3/42628. 이중우선순위큐/이중우선순위큐.py
import heapq


def solution(operations):
    q = []
    for operation in operations:
        print(q)
        tmp = operation.split(" ")
        
        # 큐에 숫자 삽입
        if tmp[0] == "I":
            heapq.heappush(q, int(tmp[1]))
            
        elif tmp[0] == "D" and q:
            
            # 최솟값 삭제
            if tmp[1] == "-1":
                heapq.heappop(q)
            
            # 최댓값 삭제
            elif tmp[1] == "1":
                q.remove(max(q))
    q.sort()
    if q:
        return [q[-1], q[0]]
    
    else:
        return [0, 0]